### PR TITLE
[MooreToCore] Add multibit DetectEventOp support

### PR DESCRIPTION
**Pull request by AndreyVV-100**
Hi! This PR add support for such code:

```verilog
module Mod(input [1:0] a, output logic b);
always @(a)
    b <= a[0];
endmodule
```

I'm not sure about correctness of this code in common case due to #7925 but now (as I understand) only [width-1:0] arrays supported in Moore dialect.

Thank you in advance!

### CODE REVIEW

**Review by fabianschuiki — APPROVED** [review]
Looks good to me! Thanks a lot for fixing this :partying_face:, really appreciated. Let me know if you'd like to merge this yourself once CI passes, or whether you'd like me to land this PR for you :slightly_smiling_face:

**Comment by AndreyVV-100** [comment]
> Looks good to me! Thanks a lot for fixing this 🥳, really appreciated. Let me know if you'd like to [merge this yourself](https://llvm.org/docs/DeveloperPolicy.html#obtaining-commit-access) once CI passes, or whether you'd like me to land this PR for you 🙂

I will be grateful if you land this PR.

**Pull request merged**
